elasticsearch_exporter.go

```diff
@@ -335,41 +94,43 @@ type Exporter struct {
 
 // NewExporter returns an initialized Exporter.
 func NewExporter(uri string, timeout time.Duration) *Exporter {
-	counters := make(map[string]*prometheus.CounterVec)
-	gauges := make(map[string]*prometheus.GaugeVec)
+	counters := make(map[string]*prometheus.CounterVec, len(counterMetrics)+len(counterVecMetrics))
+	gauges := make(map[string]*prometheus.GaugeVec, len(gaugeMetrics)+len(gaugeVecMetrics))
 
 	for name, info := range counterVecMetrics {
-		log.Printf("Registering %s", name)
+		log.Println("Registering", name)
 		counters[name] = prometheus.NewCounterVec(prometheus.CounterOpts{
 			Namespace: namespace,
 			Name:      name,
 			Help:      info.help,
-		}, append([]string{"cluster", "node"}, info.labels...))
```

Comment:
Given that one ES node (as in instance of ES running, not physical machine) can only be part of one cluster, I would prefer configuring it on the prometheus side as well. This is more explicit than having the exporter decide about those things on their own. I removed the cluster label for now.

I'm using consul as well and would use the relabling feature to set a cluster label from __meta_consul_service or __meta_consul_tags: http://prometheus.io/docs/operating/configuration/#consul-sd-configurations-consul_sd_config

@ewr Maybe that's a good/better option for you as well?